src/plugins/intel_cpu/src/scratch_pad.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "common/memory.hpp"
+#include <memory>
+#include "cpu_memory.h"
+#include "dnnl_extension_utils.h"
+
+namespace ov {
+namespace intel_cpu {
+
+class ScratchPad {
+    dnnl::engine eng;
+    Memory m_curScratchpadMem;
+
+public:
+    ScratchPad(dnnl::engine eng) : eng(eng), m_curScratchpadMem(eng) {}
+
+    void setScratchPad(std::unordered_map<int, dnnl::memory> & args, const dnnl::memory::desc & md) {
+        // Internal scratch pad memory will only increase for now
+        // Memory re-allocation is free from multi-threading safety issue
+        // as long as it's called from stream's scheduling thread
```

Comment:
Technically is not true, since we are not explicitly using a specific memory manager, just a default one instead. And this is only a coincidence that by default memory class use a memory manager with memory increase policy.

Reply:
changed to use the memory manager as underlying mechanism